### PR TITLE
Use static DateTime.UnixEpoch and RandomNumberGenerator.Fill()

### PR DESCRIPTION
**Pull request by turbedi**
- Replace a static DateTime field with DateTime.UnixEpoch
- Use static RandomNumberGenerator.Fill() method and use stackalloc

# PR Summary

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Computer.cs [hunk 1]
```diff
@@ -1822,12 +1822,9 @@ internal static string GetRandomPassword(int passwordLength)
         {
             const int charMin = 32, charMax = 122;
             const int allowedCharsCount = charMax - charMin + 1;
-            byte[] randomBytes = new byte[passwordLength];
+            byte[] randomBytes = RandomNumberGenerator.GetBytes(passwordLength);
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/Computer.cs:1825]: GetRandomPassword() is not used at all.

File: src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs [hunk 2]
```diff
@@ -1167,8 +1167,7 @@ internal static string GetTranscriptPath(string baseDirectory, bool includeDate)
             // bytes of randomness (2^48 = 2.8e14) would take an attacker about 891 years to guess
             // a filename (assuming they knew the time the transcript was started).
             // (5 bytes = 3 years, 4 bytes = about a month)
-            byte[] randomBytes = new byte[6];
-            System.Security.Cryptography.RandomNumberGenerator.Create().GetBytes(randomBytes);
+            byte[] randomBytes = System.Security.Cryptography.RandomNumberGenerator.GetBytes(6);
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs:1170]: You could use stack allocated Span.

**Comment by turbedi** [comment]
Addressed review comments

**Review by iSazonov — APPROVED** [review]

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Pull request merged**